[WIP] Fix E2E DataFrameTests for Spark 3.0.

elvaliuliuliu opened the pull request:
This fixes E2E DataFrameTests that are failing for Spark 3.0. because the APIs are either removed or deprecated.

This is a part of the effort to bring in CI for Spark 3.0: #348.

elvaliuliuliu commented:
This PR currently fixes `TestSignaturesV2_3_X`.

Moreover, not only `TestGroupedMapUdf` is failing but also `TestVectorUdf`, `TestWithColumn`, `TestUdf` in DataFrameTests for Spark 3.0. They are all failing because of `collectToPython` API.

But I am not sure how to fix `DataFrame.collect()` which is using `collectToPython`. @imback82 / @suhsteve Any suggestions? Thanks!

imback82 commented on src/csharp/Microsoft.Spark/Sql/DataFrame.cs:
```diff
@@ -714,7 +714,11 @@ public IEnumerable<Row> Collect()
+        /// This API is removed in Spark 3.0.
+        /// </remarks>
         /// <returns>Row objects</returns>
+        [Removed(Versions.V3_0_0)]
```
This is not removed. They changed the contract. We should follow this: https://github.com/apache/spark/blob/0a7095156bdb565133f7dcc74546c51a5e5d2414/python/pyspark/rdd.py#L175

imback82 commented:
> Moreover, not only `TestGroupedMapUdf` is failing but also `TestVectorUdf`, `TestWithColumn`, `TestUdf` in DataFrameTests for Spark 3.0. They are all failing because of `collectToPython` API.

Did you build the worker with the latest changes? I think I fixed it in #311.

elvaliuliuliu commented:
> > Moreover, not only `TestGroupedMapUdf` is failing but also `TestVectorUdf`, `TestWithColumn`, `TestUdf` in DataFrameTests for Spark 3.0. They are all failing because of `collectToPython` API.
> 
> Did you build the worker with the latest changes? I think I fixed it in #311.

It works for me after I updated worker. Thanks!

imback82 commented:
This is being addressed by #655.
